### DISCUSSION TITLE
Brothers DCP 1612 printer

### DISCUSSION THREAD

**Discussion by 01fiore** [mocaccinoOS/mocaccino#125] [General] [answered]
Hello everyone

I come to you because I can’t get my Brothers DCP 1612 printer to work. I installed the 1610W package but Cups via the GUI in Kde control center don’t see it. 

If anyone can help me, thank you in advance.

**Comment by msdobrescu** [comment]
@01fiore, what about the comment [here](https://github.com/mocaccinoOS/community-repository/issues/27#issuecomment-1196816484)?
The driver needs to be configured like that:

> You may need to be in the scanner or plugdev group in order to use the scanner. To add a network scanner to sane, run:
> brsaneconfig4 -a name=(name your device) model=(model name) ip=xx.xx.xx.xx
> or simply run brsaneconfig4 for more options

Did you try to follow the above steps?

**Comment by 01fiore** [comment]
Hello, it is not a problem of permissions, Cups does not find the driver, with Sabayon I used the ebuild Zugaina.
https://gpo.zugaina.org/net-print/brother-dcp1610w-bin/ChangeLog

**Comment by msdobrescu** [comment]
I see, in the [original post](https://github.com/mocaccinoOS/community-repository/issues/27), you've pointed to a [different binary](https://gpo.zugaina.org/media-gfx/brother-dcp1610w-bin).
We'll add this [one too](https://gpo.zugaina.org/net-print/brother-dcp1610w-bin) right after we finish the current C-R build cycle, which may take a day or two.

**Comment by 01fiore** [comment]
Thank you very much, and have a good day.
________________________________
De : Mihai Sorin Dobrescu ***@***.***>
Envoyé : mardi 18 avril 2023 08:32
À : mocaccinoOS/mocaccino ***@***.***>
Cc : fiore ***@***.***>; Mention ***@***.***>
Objet : Re: [mocaccinoOS/mocaccino] Brothers DCP 1612 printer (Discussion #125)


I see, in the original post<https://github.com/mocaccinoOS/community-repository/issues/27>, you've pointed to a different binary<https://gpo.zugaina.org/media-gfx/brother-dcp1610w-bin>.
We'll add this one too<https://gpo.zugaina.org/net-print/brother-dcp1610w-bin> right after we finish the current C-R build cycle, which may take a day or two.

**Comment by 01fiore** [comment]
I take this post because I have difficulties to make the printer work,There are two overlays for Gentoo and two drivers in Mocaccino, i can’t remember which one I was using with Sabayon, when I install the drivers, it installs the brother-eula GPL-2 which are not compatible, merci.

https://gpo.zugaina.org/net-print/brother-dcp1610w-bin

![Stampante](https://github.com/user-attachments/assets/6bb817ea-5849-4ddf-98f1-add16d26629a)

**Comment by msdobrescu** [comment]
The one in the link above is packaged as apps/brother-dcp1610w-bin in the community repo.

Is that what you need?

If not, would this be the one?
https://gpo.zugaina.org/Overlays/brother-overlay/media-gfx/brother-dcp1610w-bin

**Comment by 01fiore** [comment]
The packages in Mocaccino are not working, I can’t remember which is the overlay I used with Sabayon.

$ sudo luet search apps/brother
 INFO   --- Search results (apps/brother): ---                                                                                                         
> apps/brother-scan4-bin-0.4.10-r2+1
 -> Category: apps
 -> Name: brother-scan4-bin
 -> Version: 0.4.10-r2+1
 -> Description: Brother scanner tool version 4
 -> Repository: mocaccino-community-stable 
 -> Uri: http://support.brother.com/g/s/id/linux/en/index.html 
 -> Installed: false 
> apps/brother-dcp1610w-bin-brscan4-1.0-r1+1
 -> Category: apps
 -> Name: brother-dcp1610w-bin-brscan4
 -> Version: 1.0-r1+1
 -> Description: Scanner driver for Brother DCP-1610NW (brscan4)
 -> Repository: mocaccino-community-stable 
 -> Uri: http://support.brother.com 
 -> Installed: false 
> apps/brother-dcp1610w-bin-3.0.1-r2
 -> Category: apps
 -> Name: brother-dcp1610w-bin
 -> Version: 3.0.1-r2
 -> Description: Brother printer driver for DCP-1610W
 -> Repository: mocaccino-community-stable 
 -> Uri: http://support.brother.com 
 -> Installed: true 
> apps/brother-scan4-bin-0.4.10-r2+1
 -> Category: apps
 -> Name: brother-scan4-bin
 -> Version: 0.4.10-r2+1
 -> Description: Brother scanner tool version 4
 -> Repository: mocaccino-community-stable 
 -> Uri: http://support.brother.com/g/s/id/linux/en/index.html 
 -> Installed: false 
> apps/brother-dcp1610w-bin-3.0.1-r2
 -> Category: apps
 -> Name: brother-dcp1610w-bin
 -> Version: 3.0.1-r2
 -> Description: Brother printer driver for DCP-1610W
 -> Repository: mocaccino-community-stable 
 -> Uri: http://support.brother.com 
 -> Installed: true 
> apps/brother-dcp1610w-bin-brscan4-1.0-r1+1
 -> Category: apps
 -> Name: brother-dcp1610w-bin-brscan4
 -> Version: 1.0-r1+1
 -> Description: Scanner driver for Brother DCP-1610NW (brscan4)
 -> Repository: mocaccino-community-stable 
 -> Uri: http://support.brother.com 
 -> Installed: false 

I need the brother-dcp1610w package.

**Comment by msdobrescu** [comment]
According to brother, it downloads and installs the same drivers as you take from here: https://support.brother.com/g/b/downloadlist.aspx?c=eu_ot&lang=en&prod=dcp1612w_us_eu&os=128

Probably, some configuration must be done further in CUPS.

**Comment by 01fiore** [comment]
With Arch-Linux, I proceed in this way:

$ sudo pacman -S cups

$ sudo pacman -S cups-pdf

$ sudo pacman -S print-manager

$ sudo pacman -S system-config-printer

$ yay -S brother-dcp1610w

$ sudo systemctl enable cups


With mocaccino I have trouble finding the packages:

$ sudo luet search net-print/cups
INFO   --- Search results (net-print/cups): --- 

luet search net-print/cups-pdf
INFO   --- Search results (net-print/cups-pdf): ---

**Reply by msdobrescu** [reply]
The packages are not necessarily the same between distros.
In MOS, cups is in layers/X.
As long as you use a desktop, you get X layer inherently.
Let's look at the brother package in Arch.

**Comment by msdobrescu** [comment]
Hello, I have adjusted the package apps/brother-dcp1610w-bin, please try the new version.

**Reply by 01fiore** [reply]
With the new draiver, the printer installed, I printed a white-paper test-auto page with 4 black dots, but it still does not work in systemsettings:
![brother](https://github.com/user-attachments/assets/4c05a297-5662-454c-a7e3-f4c7b12d00af)
I don’t understand.

**Reply by 01fiore** [reply]
Sorry, I do not know well Mocaccino, this morning I gave:

sudo luet search brother

sudo luet uninstall apps/brother-dcp1610w-bin

sudo luet repo update

sudo luet upgrade

sudo luet install apps/brother-dcp1610w-bin

The printer has found the drivers.

Now I have given back:

sudo luet repo update

sudo luet upgrade

He updated the Brother drivers, but in systemsettings the printer is no longer found:

![brot](https://github.com/user-attachments/assets/d84c8731-fc42-4cb9-aaad-b1edca8c70e6)

Merci

**Comment by 01fiore** [comment]
Hello, i would like to solve the problem:

1)
# luet repo update
 INFO   Downloading quay.io/mocaccino/mocaccino-community:20240923072109-repository.yaml
 INFO   Pulled: sha256:89edea34c16b970db65a1437a3871509fa4d55ff15cd96303e29508bb325692a
 INFO   Size: 680B
 INFO   Downloading quay.io/mocaccino/mocaccino-community:20240923072109-tree.tar.gz
 INFO   Pulled: sha256:aebafbaa1f9f4ee5e13d9901bb64f2469d20ffbbc11cc3677863687d7e705e25
 INFO   Size: 221.2KiB
 INFO   Downloading quay.io/mocaccino/mocaccino-community:20240923072109-repository.meta.yaml.tar
 INFO   Pulled: sha256:748d2c43ae30e3b4342eaea15210206400b1bbc7744f0c266d0599a9d3886c29
 INFO   Size: 39.88MiB
 INFO   🏠  Repository mocaccino-community revision: 2289 (2024-09-23 09:21:09 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-community-stable Priority: 50 Type: docker
 INFO   Downloaded repository.yaml of 0.00 MB ( 0.00 MiB/s )
 INFO   Downloaded tree.tar.zst of 0.00 MB ( 0.00 MiB/s )
 INFO   Downloaded repository.meta.yaml.tar.zst of 0.00 MB ( 0.01 MiB/s )
 INFO   🏠  Repository luet-official revision: 1 (2023-10-25 10:10:22 +0200 CEST)
 INFO   ℹ  Repository: luet Priority: 1 Type: http
 INFO   Downloading quay.io/mocaccino/mocaccino-community:20240920191111-repository.yaml
 INFO   Pulled: sha256:7cac0d9e13a8dd8de85d442f74847038d4467fae724c4813b0e8ece963325e05
 INFO   Size: 680B
 INFO   Downloading quay.io/mocaccino/mocaccino-community:20240920191111-tree.tar.gz
 INFO   Pulled: sha256:36fef1d19ee010c48a3044b3e605ec7c6640da44b433ea254474585bae0e724c
 INFO   Size: 221KiB
 INFO   Downloading quay.io/mocaccino/mocaccino-community:20240920191111-repository.meta.yaml.tar
 INFO   Pulled: sha256:0906ab955cb1d8380e48c979b05e88acd0dbcea55f698a86e7fe255048b4a5fb
 INFO   Size: 39.84MiB
 INFO   🏠  Repository mocaccino-community revision: 2270 (2024-09-20 21:11:11 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-community-stable Priority: 50 Type: docker
 INFO   Downloading quay.io/mocaccino/desktop:20240920132256-repository.yaml
 INFO   Pulled: sha256:19e4089793dce1427d81f13cf37ff45a5f3442530ed4b2a3cdca56e1d7825d48
 INFO   Size: 680B
 INFO   Downloading quay.io/mocaccino/desktop:20240920132256-tree.tar.gz
 INFO   Pulled: sha256:19f79f8a6289970d6164e5a45a8d50744df502c78bf8e5db16c0d772dc6ec9a4
 INFO   Size: 56.84KiB
 INFO   Downloading quay.io/mocaccino/desktop:20240920132256-repository.meta.yaml.tar
 INFO   Pulled: sha256:fa4e0ed9e5fff32b4fd87269005ee9d1513395b91cc309331848259267a0b399
 INFO   Size: 120.9MiB
 INFO   🏠  Repository MocaccinoOS Desktop revision: 1340 (2024-09-20 15:22:56 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-desktop-stable Priority: 3 Type: docker
 INFO   Downloading quay.io/mocaccino/extra:repository.yaml
 INFO   Pulled: sha256:a75a4db6e35fcc29161386b56c0109e681f4694f32225a8dad5193d6896390f4
 INFO   Size: 630B
 INFO   Downloading quay.io/mocaccino/extra:tree.tar.gz
 INFO   Pulled: sha256:f09043b507ef2231c6f1e7c7475ff3a7d8a85fec697b81401b2fc3ec247a8ffd
 INFO   Size: 6.796KiB
 INFO   Downloading quay.io/mocaccino/extra:repository.meta.yaml.tar
 INFO   Pulled: sha256:069b7a8f402f5bb51c578ed898c5d74c93fc6fa7470f1449d0cb371273d57001
 INFO   Size: 744.5KiB
 INFO   🏠  Repository mocaccino-extra revision: 129 (2023-04-23 08:25:45 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-extra-stable Priority: 2 Type: docker
 INFO   Downloading quay.io/mocaccino/kernel-stable:repository.yaml
 INFO   Pulled: sha256:aa7a67ba68b8198d1ed531f49e92860adfacc6a73969e9ee627d86a8e44ed829
 INFO   Size: 616B
 INFO   Downloading quay.io/mocaccino/kernel-stable:tree.tar.gz
 INFO   Pulled: sha256:71fee119120ee983c41533486203746eda66d0c3e2b4f1abdb831690a662c7f0
 INFO   Size: 1.859KiB
 INFO   Downloading quay.io/mocaccino/kernel-stable:repository.meta.yaml.tar
 INFO   Pulled: sha256:6d5a7801493c1f2122105e15fd20455b3c191007b0c42c7df5f578475ee09269
 INFO   Size: 21.46MiB
 INFO   🏠  Repository  revision: 16 (2022-03-25 09:48:48 +0100 CET)
 INFO   ℹ  Repository: mocaccino-kernel-stable Priority: 3 Type: docker
 INFO   Downloading quay.io/mocaccino/os-commons:repository.yaml
 INFO   Pulled: sha256:5cf59f19775fb079ad701934d265a1562eb1378aeb58c577e8fb0a8c673dd740
 INFO   Size: 632B
 INFO   Downloading quay.io/mocaccino/os-commons:tree.tar.gz
 INFO   Pulled: sha256:95ef1d965faeaaf6b856c94530a27a4312be3b73ff53d9872b8e36b7d134782d
 INFO   Size: 27.78KiB
 INFO   Downloading quay.io/mocaccino/os-commons:repository.meta.yaml.tar
 INFO   Pulled: sha256:d06d36140186d44231f19372a0a39e44bd0a0f917d5199ee216789b0503636f5
 INFO   Size: 1022KiB
 INFO   🏠  Repository example-repository revision: 76 (2024-09-21 22:59:55 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-os-commons-stable Priority: 2 Type: docker
 INFO   Downloaded repository.yaml of 0.00 MB ( 0.00 MiB/s )
 INFO   Downloaded tree.tar.gz of 0.00 MB ( 0.01 MiB/s )
 INFO   Downloaded repository.meta.yaml.tar.gz of 0.00 MB ( 0.01 MiB/s )
 INFO   🏠  Repository mocaccino-repository-index revision: 1 (2024-09-24 03:47:41 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-repository-index Priority: 1 Type: http

 ####

2)
# luet upgrade
 INFO  Luet 0.32.5-gd5d21653d7313ff396c0511af402834c89974ee3 2022-06-07
 INFO  Copyright (C) 2019-2022 Ettore Di Giacinto
       This program comes with ABSOLUTELY NO WARRANTY.
       This is free software, and you are welcome to redistribute it under certain conditions.
       For documentation, visit https://luet.io.
 INFO   :::> Upgrade
 INFO   🤔  Computing upgrade, please hang tight... 💤
 INFO   ⚡  Proposed version changes to the system:


Old version                                          | New version                                          | License
repository/mocaccino-community-stable-20240923072109 | repository/mocaccino-community-stable-20240923203314 |

 INFO   By going forward, you are also accepting the licenses of the packages that you are going to install in your system.
 INFO   Do you want to continue with this operation? [y/N]:
y
Downloading packages [0/1] █                                                                                                                    0% | 1s
mocaccino-community-stable-repository-20240923203314.package.tar.gz [1/372] █                                                                   0% | 0s
 INFO   Downloaded mocaccino-community-stable-repository-20240923203314.package.tar.gz of 0.00 MB ( 0.00 MiB/s )

Downloading packages [1/1] ██████████████████████████████████████████████████████████████████████████████████████████████████████████████████ 100% | 1s
mocaccino-community-stable-repository-20240923203314.package.tar.gz [372/372] ███████████████████████████████████████████████████████████████ 100% | 0s
 INFO   Checking for file conflicts..
 INFO   Done checking for file conflicts..
 INFO   Found protected file /etc/luet/repos.conf.d/mocaccino-community-stable.yml. Creating /etc/luet/repos.conf.d/._cfg0001_mocaccino-community-stable.yml.
 INFO   📦  Package  repository/mocaccino-community-stable-20240923203314 installed
moca /home/fiore # luet search brother
 INFO   Downloading quay.io/mocaccino/mocaccino-community:20240923203314-repository.yaml
 INFO   Pulled: sha256:89164f9d43b0848d91cbc063052e708a5f2d0cb453a0ee1d1bc41c5a68781aa2
 INFO   Size: 680B
 INFO   Downloading quay.io/mocaccino/mocaccino-community:20240923203314-tree.tar.gz
 INFO   Pulled: sha256:d292c825452bba9b3caa60ed6b0912c07dab54ff44dba2227f04157679411b85
 INFO   Size: 222KiB
 INFO   Downloading quay.io/mocaccino/mocaccino-community:20240923203314-repository.meta.yaml.tar
 INFO   Pulled: sha256:6570192646aa3161f23727fd030f7b5dc1e2212b9e58b7c968d7ee705cc9a5c6
 INFO   Size: 40.01MiB
 INFO   🏠  Repository mocaccino-community revision: 2293 (2024-09-23 22:33:14 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-community-stable Priority: 50 Type: docker
 INFO   --- Search results (brother): ---
> apps/brother-scan4-bin-0.4.10-r2+1
 -> Category: apps
 -> Name: brother-scan4-bin
 -> Version: 0.4.10-r2+1
 -> Description: Brother scanner tool version 4
 -> Repository: mocaccino-community-stable
 -> Uri: http://support.brother.com/g/s/id/linux/en/index.html
 -> Installed: false
> apps/brother-dcp1610w-bin-3.0.1-r2+2
 -> Category: apps
 -> Name: brother-dcp1610w-bin
 -> Version: 3.0.1-r2+2
 -> Description: Brother printer driver for DCP-1610W
 -> Repository: mocaccino-community-stable
 -> Uri: http://support.brother.com
 -> Installed: false
> apps/brother-dcp1610w-bin-brscan4-1.0-r1+1
 -> Category: apps
 -> Name: brother-dcp1610w-bin-brscan4
 -> Version: 1.0-r1+1
 -> Description: Scanner driver for Brother DCP-1610NW (brscan4)
 -> Repository: mocaccino-community-stable
 -> Uri: http://support.brother.com
 -> Installed: false
> apps/brother-dcp1610w-bin-3.0.1-r2+2
 -> Category: apps
 -> Name: brother-dcp1610w-bin
 -> Version: 3.0.1-r2+2
 -> Description: Brother printer driver for DCP-1610W
 -> Repository: mocaccino-community-stable
 -> Uri: http://support.brother.com
 -> Installed: false
> apps/brother-scan4-bin-0.4.10-r2+1
 -> Category: apps
 -> Name: brother-scan4-bin
 -> Version: 0.4.10-r2+1
 -> Description: Brother scanner tool version 4
 -> Repository: mocaccino-community-stable
 -> Uri: http://support.brother.com/g/s/id/linux/en/index.html
 -> Installed: false
> apps/brother-dcp1610w-bin-brscan4-1.0-r1+1
 -> Category: apps
 -> Name: brother-dcp1610w-bin-brscan4
 -> Version: 1.0-r1+1
 -> Description: Scanner driver for Brother DCP-1610NW (brscan4)
 -> Repository: mocaccino-community-stable
 -> Uri: http://support.brother.com
 -> Installed: false


 #####
 
3) 
 # luet install apps/brother-dcp1610w-bin
 INFO  Luet 0.32.5-gd5d21653d7313ff396c0511af402834c89974ee3 2022-06-07
 INFO  Copyright (C) 2019-2022 Ettore Di Giacinto
       This program comes with ABSOLUTELY NO WARRANTY.
       This is free software, and you are welcome to redistribute it under certain conditions.
       For documentation, visit https://luet.io.
 INFO   :::> Install
 INFO   🤔  Checking for available upgrades
 INFO   Nothing to upgrade
 INFO   Packages that are going to be installed in the system:

Program Name                         | Version    | License            | Repository
apps/brother-dcp1610w-bin-3.0.1-r2+2 | 3.0.1-r2+2 | brother-eula GPL-2 | mocaccino-community-stable

 INFO   By going forward, you are also accepting the licenses of the packages that you are going to install in your system.
 INFO   Do you want to continue with this operation? [y/N]:
y
 INFO   Checking for file conflicts..
▀   INFO   Done checking for file conflicts..
 INFO   📦  Package  apps/brother-dcp1610w-bin-3.0.1-r2+2 installed
 INFO   Executing finalizer for apps/brother-dcp1610w-bin-3.0.1-r2+2
 INFO   🐚  Executing finalizer on  / sh [-c glib-compile-schemas /usr/share/glib-2.0/schemas > /dev/null 2>&1]
 INFO
 INFO   🐚  Executing finalizer on  / sh [-c ldconfig]
 INFO
 INFO   🐚  Executing finalizer on  / sh [-c env-update && source /etc/profile]
 INFO   >>> Regenerating /etc/ld.so.cache...
 
 4)
 
 The problem persists, either the PPD present in Luet are not functional or something is missing in the configuration of Cups:
![Cups](https://github.com/user-attachments/assets/56161a51-aef6-44d3-b41c-31f4b1d120ce)

**Comment by msdobrescu** [comment]
Me too, but I have no such printer and it is difficult to understand what it needs from the drivers point of view.
I will try another configuration and let's see what is the outcome.
I'll let you know when to upgrade once more and see. OK?

**Comment by 01fiore** [comment]
Ok merci.

**Comment by msdobrescu** [comment, marked as answer]
I have reworked the package a bit, please give it a try.

**Comment by 01fiore** [comment]
Excellent news, the printer works perfectly, bravo.

**Comment by msdobrescu** [comment]
Glad to hear that! Thanks for your patience and collaboration!
